src/main/java/com/ankoma88/datastructures/Performance/service/PerformanceAnalysisService.java

```diff
@@ -0,0 +1,186 @@
+package com.ankoma88.datastructures.performance.service;
+
+import com.ankoma88.datastructures.performance.data.entities.MeasurementRecord;
+import com.ankoma88.datastructures.performance.data.repository.MeasurementsRepository;
+import com.ankoma88.datastructures.performance.domain.performance.PerformanceAnalysis;
+import com.ankoma88.datastructures.performance.domain.model.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static com.ankoma88.datastructures.performance.domain.model.DataStructureEnum.ARRAYLIST;
+import static com.ankoma88.datastructures.performance.domain.model.DataStructureEnum.LINKEDLIST;
+import static com.ankoma88.datastructures.performance.domain.model.OperationEnum.*;
+import static com.ankoma88.datastructures.performance.domain.model.ScenarioEnum.*;
+
+@Service
+public class PerformanceAnalysisService implements AnalysisService {
+
+    private static final Logger LOGGER = Logger.getLogger(PerformanceAnalysisService.class.getName());
+
+    @Autowired
+    @Qualifier("measurementsTaskExecutor")
+    private Executor measurementsThreadPool;
+
+    @Autowired
+    private MeasurementsRepository measurementsRepository;
+
+    @Override
+    public List<PerformanceMeasurementDto> getLatestMeasurements(int size){
+        PageRequest pageRequest = PageRequest.of(0, size);
+        Page<MeasurementRecord> records = measurementsRepository.getMeasurementRecords(pageRequest);
+        List<MeasurementRecord> measurementRecords = records.getContent();
+        return buildMeasurements(measurementRecords);
+    }
+
+    @Override
+    public PerformanceDto measureOperations(int listSize, int repeats) {
```

Comment:
method is too complex and lengthy. Refactor into smaller methods for better readability and maintainability.